src/typescript/typescript-parser.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as ts from 'typescript';
+
+import {Analyzer} from '../analyzer';
+import {Parser} from '../parser/parser';
+
+import {ParsedTypeScriptDocument} from './typescript-document';
+
+const _compilerOptions: ts.CompilerOptions = {
+  allowJs: true,
+};
+
+export class TypeScriptParser implements Parser<ParsedTypeScriptDocument> {
+  private _analyzer: Analyzer;
+
+  constructor(analyzer: Analyzer) {
+    this._analyzer = analyzer;
+  }
+
+  parse(contents: string, url: string): ParsedTypeScriptDocument {
+    const host = new AnalyzerCompilerHost(
+        this._analyzer, new Map([[url, contents], ['lib.ts', '']]));
+    const program = ts.createProgram([url], _compilerOptions, host);
+    return new ParsedTypeScriptDocument({
+      url,
+      contents,
+      program,
+      locationOffset: undefined,
+      // TODO(justinfagnani): Fix or remove generics on ParsedDocument that
+      // make this 1) necessary 2) possible (not throwing)
+      ast: null as any as ts.Node,
```

Comment:
I wonder if we could specify that ast on ParsedTypeScriptDocument is of type `null` or `Program` manually. It's nice that we get the typings of stuff like _sourceRangeForNode